### PR TITLE
Navigation with name and course name

### PR DESCRIPTION
**Pull request by pandafulmanda**
## Student
screen shot 2015-05-06 at 1 55 39 am

## Teacher
screen shot 2015-05-06 at 1 56 59 am

## Hamburger
screen shot 2015-05-06 at 2 09 01 am

Todo
- [x] separate nav component out from app
- [x] current user store loads name
  - [x] includes a backend update [tutor-server/#245](https://github.com/openstax/tutor-server/pull/245)
  - [x] Possibly refactor nav username methods into mixin
- [x] course store.get course id in nav for course name
  - [x] adjust course links based on roles
- [x] styling/layout
  - [x] css refactoring
  - [x] small screens?
- [x] tests

### CODE REVIEW

File: api/user.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  name: "Penelope Morar"
```
Comment (philschatz) [inline api/user.json]: This seems to be invalid JSON. Also, who's `Penelope Morar`?

Comment (nathanstitt) [inline api/user.json]: Oops..  Right you are.  It was just copied and pasted from the back-end from a console.log, hence the invalid json.  Fixing up now.

File: src/components/nav.cjsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+  componentWillMount: ->
+    unless @state.name
+      @_addListener()
+      CurrentUserActions.loadName()
+
+  render: ->
+    <p className="navbar-text user-name">{@state.name}</p>
+
+  ## These methods are all copied from Loadable.  Extract into mixin?
```
Comment (pandafulmanda) [inline src/components/nav.cjsx]: Okay, heads up, I am doing this.

**Comment by philschatz** [comment]
I'm a little worried about keeping the store in sync (with the "Current Course"). What about https://github.com/openstax/tutor-js/pull/182/files#r29657707 so this PR can still rely on the `courseId` in the router?

**Pull request merged**